pkg/runtime/secretstores/registry.go

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2021 Layotto Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package secretstores
+
+import (
+	"github.com/dapr/components-contrib/secretstores"
+	"mosn.io/layotto/components/pkg/info"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+const ServiceName = "secretStore"
+
+type (
+
+	// Registry is used to get registered secret store implementations.
+	Registry interface {
+		Register(ss ...*SecretStoresFactory)
+		Create(name, version string) (secretstores.SecretStore, error)
+	}
+
+	secretStoreRegistry struct {
+		secretStores map[string]func() secretstores.SecretStore
+		info         *info.RuntimeInfo
+	}
+)
+
+// NewRegistry returns a new secret store registry.
+func NewRegistry(info *info.RuntimeInfo) Registry {
+	info.AddService(ServiceName)
+	return &secretStoreRegistry{
+		secretStores: map[string]func() secretstores.SecretStore{},
+		info:         info,
+	}
+}
+
+// Register adds one or many new secret stores to the registry.
+func (s *secretStoreRegistry) Register(ss ...*SecretStoresFactory) {
+	for _, component := range ss {
+		s.secretStores[createFullName(component.Name)] = component.FactoryMethod
+		s.info.RegisterComponent(ServiceName, component.Name)
+	}
+}
+
+// Create instantiates a secret store based on `name`.
+func (s *secretStoreRegistry) Create(name, version string) (secretstores.SecretStore, error) {
+	if method, ok := s.getSecretStore(name, version); ok {
+		s.info.LoadComponent(ServiceName, name)
+		return method(), nil
+	}
+
+	return nil, errors.Errorf("couldn't find secret store %s/%s", name, version)
+}
+
+func (s *secretStoreRegistry) getSecretStore(name, version string) (func() secretstores.SecretStore, bool) {
+	nameLower := strings.ToLower(name)
+	versionLower := strings.ToLower(version)
+	secretStoreFn, ok := s.secretStores[nameLower+"/"+versionLower]
+	if ok {
+		return secretStoreFn, true
+	}
+	if versionLower == "" || versionLower == "v0" || versionLower == "v1" {
+		secretStoreFn, ok = s.secretStores[nameLower]
+	}
+	return secretStoreFn, ok
+}
+
+func createFullName(name string) string {
+	return strings.ToLower("secretstores." + name)
```

Comment:
It seems that the storeName will be modified automatically. So what `storeName` should users pass when invoking secret API ?
Could you add an end-to-end demo that starts a layotto and invoke secret API to show users how to use it?

Reply:
1. This is basically looking at the way DAPR thinks,Leave extension points. The default implementation is the SecretStore Register, but users can also customize their own extension points.  [dapr方案](https://github.com/dapr/dapr/pull/930)
2. I can

Reply:
I just read the corrosponding code in Dapr. Dapr add a "secretstores." prefix for every component name because the configuration is like "secretstores.local.file":
![image](https://user-images.githubusercontent.com/26001097/152955453-6e58a76e-bc60-4417-9c0b-7e3ae0a70428.png)
As for Layotto , users write "local.file" instead of "secretstores.local.file". So the `createFullName` function is not needed. I will delete it in your branch :)